sidecar/internal/csiaddonsnode/csiaddonsnode_test.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2021 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csiaddonsnode
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	csiaddonsv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/api/v1alpha1"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func Test_lookupEnv(t *testing.T) {
+
+	t.Run("Env Var set", func(t *testing.T) {
+		key := "hello"
+		value := "world"
+		defer setEnvVar(t, key, value)()
+
+		got, err := lookupEnv(key)
+		if (err != nil) != false {
+			t.Errorf("lookupEnv() error = %v, wantErr %v", err, false)
+			return
+		}
+		if got != value {
+			t.Errorf("lookupEnv() = %v, want %v", got, value)
+		}
+	})
+
+	t.Run("Env Var not set", func(t *testing.T) {
+		key := "hello"
+		_, err := lookupEnv(key)
+		if err == nil {
+			t.Errorf("lookupEnv() error = %v, wantErr %v", err,
+				fmt.Errorf("Required environemental variable %q not found", key))
+		}
+	})
+}
+
+func Test_getCSIAddonsNode(t *testing.T) {
+	var (
+		podName     = "pod"
+		podNamspace = "default"
+		podUID      = "123"
+		podIP       = "110"
+	)
+	defer setEnvVar(t, podNameEnvKey, podName)()
```

Comment:
try using https://pkg.go.dev/testing#T.Setenv instead of implementing own logic to set and unset env